### DISCUSSION TITLE
JSON sending data

### DISCUSSION THREAD

**Discussion by VladimirVecera** [arendst/Tasmota#18860] [Q&A]
I have a Tasmota V2 socket and I need to send the measured data to my website. Can tasmota's POST method send JSON data? If so, how? Thank you very much.

**Comment by s-hadinger** [comment]
I have no idea what a Tasmota V2 socket is. 

Short answer: yes with ESP32 or ESP32C3, no with ESP8266 (at least not securely)

Best option is to use MQTT

**Reply by VladimirVecera** [reply]
This is tasmota socket: [https://www.tasmota.shop/cs/zasuvky/5-zasuvka-wifi-tasmota-v2.html](url)
I don't want MQTT. I need to get the measured data about the consumption in watts and send it to a PHP file on the web hosting, where I want to store the data in mySQL and make graphs.
How can tasmota create JSON data for me with POST by sending the data to a URL?

**Reply by barbudor** [reply]
Be aware that this shop is in no way endorsed by or related to the Tasmota open source project
Although Tasmota is OSS and the logo is, I'm not sure that using "Tasmota" as a trademark like this, which may confuse people, is really allowed. Generally vendors just sell devices on their brand advertising "pre-loaded with Tasmota"


Regarding your question, you should look at Rules to trigger on sensors data ad then WeqQuery to make a HTTP GET or POST
AS pointed by @sfromis, on ESP8266 only HTTP is supported while on ESP32 HTTPS is supported too.
